src/storage.rs

```diff
@@ -1,85 +1,112 @@
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 use serde_json::{from_str, to_string};
+use std::fmt;
 use std::fs::{File, OpenOptions};
-use std::io::{Read, Write, Seek, SeekFrom};
+use std::io::{Read, Seek, SeekFrom, Write};
 use std::ops::{Deref, DerefMut};
-use std::fmt;
 
+// General traits
+pub trait StorageReadGuard<'a, T>: Deref<Target = T> {}
+pub trait StorageWriteGuard<'a, T>: DerefMut<Target = T> {}
+
+pub trait Storage<'a>: fmt::Display {
```

Comment:
Why is this lifetime required? I understand that the lifetimes below are needed to guarantee that the guards live as long as the storage.

Reply:
Does this work for you? It works locally for me, but I am just compiling storage.rs because the rest doesn't compile because of the protobuf/serde stuff in the sdk.

https://github.com/aludvik/sawtooth-pbft-1/tree/state-caching-no-lifetime